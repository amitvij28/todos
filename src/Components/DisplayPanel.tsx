import React, { useContext } from "react";
import { Grid } from "@material-ui/core";
import Lane from "./Lane";
import { todos, TodoStatus } from "../Context/TodoProvider";
import { HTML5Backend } from "react-dnd-html5-backend";
import { DndProvider } from "react-dnd";

const DisplayPanel = () => {
    const { state } = useContext(todos);

    return (
        <DndProvider backend={HTML5Backend}>
            <Grid container spacing={2}>
                <Grid item xs={4}>
                    <Lane
                        todoList={state.todos.filter(
                            (t) => t.status === TodoStatus.tbd
                        )}
                        heading="To Be Done"
                        type={TodoStatus.tbd}
                    />
                </Grid>
                <Grid item xs={4}>
                    <Lane
                        todoList={state.todos.filter(
                            (t) => t.status === TodoStatus.doing
                        )}
                        heading="Doing"
                        type={TodoStatus.doing}
                    />
                </Grid>
                <Grid item xs={4}>
                    <Lane
                        todoList={state.todos.filter(
                            (t) => t.status === TodoStatus.done
                        )}
                        heading="Done"
                        type={TodoStatus.done}
                    />
                </Grid>
            </Grid>
        </DndProvider>
    );
};

export default DisplayPanel;
