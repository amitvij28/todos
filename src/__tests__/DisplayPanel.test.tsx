import React from "react";
import { shallow } from "enzyme";
import DisplayPanel from "../Components/DisplayPanel";

describe("Display Panel", () => {
    const component = shallow(<DisplayPanel />);
    it("should render", () => {
        expect(component).toMatchSnapshot();
    });
});
