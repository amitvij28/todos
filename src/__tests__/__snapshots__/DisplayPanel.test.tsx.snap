// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Panel should render 1`] = `
<DndProvider
  backend={[Function]}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <Lane
        heading="To Be Done"
        todoList={Array []}
        type="To Be Done"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <Lane
        heading="Doing"
        todoList={Array []}
        type="Doing"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={4}
    >
      <Lane
        heading="Done"
        todoList={Array []}
        type="Done"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</DndProvider>
`;
